aodndata/moorings/classifiers.py

```diff
@@ -324,9 +326,13 @@ def dest_path(cls, input_file):
             dir_list += ['ASFS', 'SOFS', cat, 'Real-time', rt_folder_name]
         elif subfac in ('SOTS', 'ASFS'):
             dir_list.append('SOTS')
-            dir_list.append(cls._get_deployment_year(input_file))
-            if cls._is_realtime(input_file):
-                dir_list.append('real-time')
+            # Check for data_product_type attribute
+            if cls._get_nc_att(input_file, 'data_product_type'):
```

Comment:
As it is, `cls._get_nc_att` will raise an error if the attribute is missing. What you want here is for it to return `False` in that case, which you can achieve like this:
```python
         if cls._get_nc_att(input_file, 'data_product_type', default=False):
```
See the docstring for `_get_nc_att`: https://github.com/aodn/python-aodncore/blob/d78192d877e3a07d7a8df178cc45ed502a22bb8f/aodncore/pipeline/fileclassifier.py#L75

Reply:
You will also need to save and check the value of the attribute -- see https://github.com/aodn/PO-Backlog/issues/3620#issuecomment-2164241924